controllers/metal3machine_controller.go

```diff
@@ -155,20 +157,23 @@ func (r *Metal3MachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		err := machineMgr.RemovePauseAnnotation(ctx)
 		if err != nil {
 			machineLog.Info("failed to check pause annotation on associated bmh")
+			conditions.MarkFalse(capm3Machine, infrav1.AssociateBMHCondition, infrav1.PauseAnnotationRemoveFailedReason, clusterv1.ConditionSeverityInfo, "")
 			return ctrl.Result{}, nil
 		}
 	} else {
 		// set pause annotation on associated bmh (if any)
 		err := machineMgr.SetPauseAnnotation(ctx)
 		if err != nil {
 			machineLog.Info("failed to set pause annotation on associated bmh")
+			conditions.MarkFalse(capm3Machine, infrav1.AssociateBMHCondition, infrav1.PauseAnnotationSetFailedReason, clusterv1.ConditionSeverityInfo, "")
 			return ctrl.Result{}, nil
 		}
 	}
 
 	// Return early if the M3Machine or Cluster is paused.
 	if annotations.IsPaused(cluster, capm3Machine) {
 		machineLog.Info("reconciliation is paused for this object")
+		conditions.MarkFalse(capm3Machine, infrav1.AssociateBMHCondition, infrav1.Metal3MachinePausedReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
should not the condition marked as True after this?

Reply:
I did not quite get what you are asking. 
If annotations Paused, then AssociateBMHCondition set to false, and the reason of the Condition set to Metal3MachinePausedReason.

Reply:
We are setting condition true [here](https://github.com/Nordix/cluster-api-provider-metal3/blob/704b5f3aef3f380f625a9a4d496249a8a2db562b/baremetal/metal3machine_manager.go#L1430) inside SetProviderID function

Reply:
it is set for `KubernetesNodeReadyCondition` but above is `AssociateBMHCondition`

Reply:
Currently there are two conditions exist in the provider:

1. `AssociateBMHCondition`
2. `KubernetesNodeReadyCondition`

First one (AssociateBMHCondition) deals with all the things concerned with BMH<->M3M association, and the latter (KubernetesNodeReadyCondition) deals with M3M <-> Machine (k8s node) part. 

This PR just sets the appropriate reasons based on different situations and according to the above logic, which object  we dealing  with at specific point. 

Concerning the conditions marked as True for the `AssociateBMHCondition`: 

1. reconciliation starts with `Reconcile` loop in the controller with condition marked as False but at the end it calls `ReconcileNormal`, where  we mark `AssociateBMHCondition` as True. 

2. `KubernetesNodeReady `: marked as True inside the SetProviderID call, which is the last one to be called after `KubernetesNodeReady` marked as False.